sdk/monitor/monitor-opentelemetry-exporter/src/utils/spanUtils.ts

```diff
@@ -379,3 +391,77 @@ export function readableSpanToEnvelope(span: ReadableSpan, ikey: string): Envelo
     },
   };
 }
+
+/**
+ * Span Events to Azure envelopes parsing.
+ * @internal
+ */
+export function spanEventsToEnvelopes(span: ReadableSpan, ikey: string): Envelope[] {
+  let envelopes: Envelope[] = [];
+  if (span.events) {
+    span.events.forEach((event: TimedEvent) => {
+      let baseType: "ExceptionData" | "MessageData";
+      const sampleRate = 100;
+      let time = new Date(hrTimeToMilliseconds(event.time));
+      let name = "";
+      let baseData: TelemetryExceptionData | MessageData;
+      const properties = createPropertiesFromSpanAttributes(event.attributes);
+      const tags: Tags = createGenericTagsFromSpan(span);
+
+      if (event.name == "exception") {
+        name = "Microsoft.ApplicationInsights.Exception";
+        baseType = "ExceptionData";
+        let typeName = "";
+        let message = "Exception";
+        let stack = "";
+        let hasFullStack = false;
+        if (event.attributes) {
+          typeName = String(event.attributes[SemanticAttributes.EXCEPTION_TYPE]);
+          stack = String(event.attributes[SemanticAttributes.EXCEPTION_STACKTRACE]);
+          if (stack) {
+            hasFullStack = true;
+          }
+          let exceptionMsg = event.attributes[SemanticAttributes.EXCEPTION_MESSAGE];
+          if (exceptionMsg) {
+            message = String(exceptionMsg);
+          }
+        }
+        let exceptionDetails: TelemetryExceptionDetails = {
+          typeName: typeName,
+          message: message,
+          stack: stack,
+          hasFullStack: hasFullStack,
+        };
+        let exceptionData: TelemetryExceptionData = {
+          exceptions: [exceptionDetails],
+          version: 2,
+          properties: properties,
```

Comment:
Do we want to populate the properties with the exception attributes if they are being reflected in the other fields? @trask Thoughts on this?

Reply:
it looks like we don't currently capture any custom properties on exceptions (https://github.com/microsoft/ApplicationInsights-Java/blob/71d610c7d2b5e7cd062bdfd3f26937b37ecb8985/agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/SpanDataMapper.java#L789)

if we did (and we should), I would apply the same deny-list of attributes to populate: https://github.com/microsoft/ApplicationInsights-Java/blob/71d610c7d2b5e7cd062bdfd3f26937b37ecb8985/agent/azure-monitor-exporter/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/SpanDataMapper.java#L127-L138